src/aof.zig

```diff
@@ -16,42 +17,24 @@ const Header = vsr.Header;
 const Client = vsr.ClientType(StateMachine, MessageBus);
 const log = std.log.scoped(.aof);
 
-const magic_number: u128 = 312960301372567410560647846651901451202;
-
-/// On-disk format for AOF Metadata.
-pub const AOFEntryMetadata = extern struct {
-    primary: u64,
-    replica: u64,
-    // Use large padding here to align the message itself to the sector boundary.
-    reserved: [4064]u8 = std.mem.zeroes([4064]u8),
-
-    comptime {
-        assert(stdx.no_padding(AOFEntryMetadata));
-    }
-};
+const magic_number: u128 = 980549605906083105481957035548671106;
 
 pub const AOFEntry = extern struct {
     /// In case of extreme corruption, start each entry with a fixed random integer,
     /// to allow skipping over corrupted entries.
     magic_number: u128 = magic_number,
 
-    /// Arbitrary metadata we want to record
-    metadata: AOFEntryMetadata,
-
-    /// The main Message to log. The actual length of the entire payload will be sector
-    /// aligned, so we might write past what the VSR header in here indicates.
-    message: [constants.message_size_max]u8 align(constants.sector_size),
+    /// The main Message to log. This is written _without_ O_DIRECT, so sector alignment is not
+    /// a concern.
+    message: [constants.message_size_max]u8 align(16),
 
     comptime {
         assert(stdx.no_padding(AOFEntry));
     }
 
-    /// Calculate the actual length of the AOFEntry that needs to be written to disk,
-    /// accounting for sector alignment.
+    /// Calculate the actual length of the AOFEntry that needs to be written to disk.
     pub fn calculate_disk_size(self: *AOFEntry) u64 {
-        const unaligned_size = @sizeOf(AOFEntry) - self.message.len + self.header().size;
-
-        return vsr.sector_ceil(unaligned_size);
+        return @sizeOf(AOFEntry) - self.message.len + self.header().size;
```

Comment:
When we read an entry in the iterator in `next`, check that the original `bytes_read` is at least sizeOf(Header).